### DISCUSSION TITLE
Options for building msi installer on Linux in v2

### DISCUSSION THREAD

**Discussion by akhileshshastri-bpi** [tauri-apps/tauri#11766] [Q&A] [answered]
The [docs](https://v2.tauri.app/distribute/windows-installer/) mention that an `.msi` installer can only be created on Windows, and that cross-platform compilation is not supported.

Can I make use of `msitools` to create the installer on Linux?

https://wiki.gnome.org/msitools

If not, are there plans to add support for cross-platform creation of `.msi` installers in the near future?

**Comment by FabianLars** [comment]
> Can I make use of msitools to create the installer on Linux?

Yes, but that's totally up to you. You cannot re-use anything of tauri's msi tooling for this.

> If not, are there plans to add support for cross-platform creation of .msi installers in the near future?

We would like to have support for it but nothing is planned for the near future. Of course we wouldn't turn down community contributions.

**Reply by akhileshshastri-bpi** [reply]
Okay thanks a lot. Do you think it will be a big task to get `msitools` working with Tauri? I ask because if so, I want to explore other options such as Azure/Gitlab pipelines or even a VM first.

**Reply by FabianLars** [reply, marked as answer]
I personally would go with github ci or a vm (if setup.exe doesn't work for you).

You _can_ try to use tauri's main.wxs file, replacing the values wrapped in `{{}}` with the actual values and see if msitools can compile that.

**Reply by FabianLars** [reply]
But if not even RemoveFile works https://gitlab.gnome.org/GNOME/msitools/-/issues/47 i doubt it can handle RemoveFolder and RemoveSymlink so i doubt it can compile our wxs file.

**Reply by akhileshshastri-bpi** [reply]
Got it, thanks for the help
